File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function (s) {
  let unique = {};

  for (var i = 0; i < s.length; i++) {
    if (unique[s[i]]) {
      unique[s[i]] += 1
    } else unique[s[i]] = 1;
  }

  let check = null;
  for (let prop in unique) {
    if (unique[prop] == 1) {
      check = prop;
      break;
    }
  }

  if (check) {
    return s.split('').findIndex(i => i == check)
  }
  return -1;
}

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.go
func firstUniqChar(s string) int {
	for i := 0; i < len(s); i++ {
		check := true
		for j := 0; j < len(s); j++ {
			if s[i] == s[j] && i != j {
				check = false
				break
			}
		}

		if check {
			return i
		}
	}
	return -1
}
